addons/portal/controllers/mail.py

```diff
@@ -172,6 +175,39 @@ def portal_chatter_init(self, res_model, res_id, domain=False, limit=False, **kw
             }
         }
 
+    def _prepare_portal_message_fetch_group_by_day(self, messages, message_ids):
+        today = fields.Date.context_today(message_ids)
+        def add_message_to_grouped_list(msg_groups, message):
+            message_id = message_ids.browse(message['id'])
+            message_date = message_id.date.date()
+            if message_date not in msg_groups.keys():
+                msg_groups[message_date] = {
+                    'label': _("Today") if message_date == today else message_date.strftime("%B %-d, %Y"),
+                    'date': message_date,
+                    'messages': [message]
+                }
+            else:
+                msg_groups[message_date]['messages'].append(message)
+            return msg_groups
+        return list(reduce(add_message_to_grouped_list, messages, {}).values())
```

Comment:
Beware of the time localisation. What if an indian post a message at 23 pm indian time on a belgian odoo server?
and isn't it more easy to read to make a simple loop to group the messages ? Please note we are stupid devs at Odoo, the simplest the better :D 
Missing docstring anyway :)

same for the next method